Is this actually working? When I run `flake8` in the SymPy repo, it reports a lot of errors in various files (I guess we have only ever run it on `sympy/`). But when I do the same with ruff, it doesn't report anything. Here's the errors flake8 reports:

```
./.ci/parse_durations_log.py:18:13: E722 do not use bare 'except'
./bin/ask_update.py:19:1: F401 'pprint' imported but unused
./bin/mailmap_check.py:17:1: F401 'os' imported but unused
./bin/mailmap_check.py:31:1: F401 'sympy' imported but unused
./bin/sympy_time.py:35:1: F401 'sympy.*' imported but unused
./bin/sympy_time_cache.py:3:1: F401 'time' imported but unused
./bin/sympy_time_cache.py:30:5: F811 redefinition of unused 'time' from line 3
./bin/sympy_time_cache.py:125:1: F401 'sympy.*' imported but unused
./bin/test_executable.py:8:1: F401 'sys' imported but unused
./bin/test_import.py:7:1: F401 'sympy' imported but unused
./bin/test_py2_import.py:18:5: F401 'sympy' imported but unused
./doc/api/conf.py:13:1: F401 'sys' imported but unused
./doc/ext/docscrape.py:10:1: F401 'sys' imported but unused
./doc/ext/docscrape.py:468:13: F841 local variable 'e' is assigned to but never used
./doc/ext/docscrape.py:484:9: F841 local variable 'signature' is assigned to but never used
./doc/ext/docscrape_sphinx.py:1:1: F401 'sys' imported but unused
./doc/ext/docscrape_sphinx.py:7:1: F401 'collections' imported but unused
./doc/ext/numpydoc.py:53:19: F821 undefined name 'unicode'
./doc/src/conf.py:421:5: E722 do not use bare 'except'
./examples/advanced/autowrap_integrators.py:66:5: F841 local variable 'B' is assigned to but never used
./examples/advanced/autowrap_ufuncify.py:76:9: F841 local variable 'plot1' is assigned to but never used
./examples/advanced/curvilinear_coordinates.py:12:1: F401 'sympy.var' imported but unused
./examples/advanced/curvilinear_coordinates.py:12:1: F401 'sympy.trigsimp' imported but unused
./examples/advanced/fem.py:62:32: F821 undefined name 'binomial'
./examples/advanced/fem.py:63:34: F821 undefined name 'binomial'
./examples/advanced/fem.py:102:32: F823 local variable 'y' defined in enclosing scope on line 21 referenced before assignment
./examples/advanced/gibbs_phenomenon.py:16:1: F401 'sympy.seterr' imported but unused
./examples/advanced/gibbs_phenomenon.py:16:1: F401 'sympy.I' imported but unused
./examples/advanced/gibbs_phenomenon.py:16:1: F401 'sympy.sign' imported but unused
./examples/advanced/gibbs_phenomenon.py:16:1: F401 'sympy.S' imported but unused
./examples/advanced/qft.py:50:13: F821 undefined name 'zeros'
./examples/advanced/qft.py:66:13: F821 undefined name 'zeros'
./examples/advanced/qft.py:106:5: F841 local variable 'mu' is assigned to but never used
./examples/advanced/relativity.py:84:9: F841 local variable 'x' is assigned to but never used
./examples/advanced/relativity.py:140:5: F841 local variable 'C' is assigned to but never used
./examples/all.py:46:1: F401 'sympy' imported but unused
./examples/all.py:150:5: E722 do not use bare 'except'
./examples/beginner/plotting_nice_plot.py:8:1: F401 'sympy.Symbol' imported but unused
./examples/beginner/precision.py:8:1: F401 'sympy' imported but unused
./isympy.py:308:13: F401 'IPython' imported but unused
./release/authors.py:3:1: F401 'os' imported but unused
./release/authors.py:122:13: F841 local variable 'parents' is assigned to but never used
./release/authors.py:132:14: F821 undefined name 'red'
./release/build_docs.py:5:1: F401 'os.path.basename' imported but unused
./release/build_docs.py:5:1: F401 'os.path.normpath' imported but unused
./release/github_release.py:239:20: F821 undefined name 'getpass'
./release/github_release.py:325:16: F821 undefined name 'HTTPBasicAuth'
./release/github_release.py:359:37: F821 undefined name 'stat'
./release/github_release.py:359:52: F821 undefined name 'stat'
./setup.py:210:9: E722 do not use bare 'except'
./setup.py:305:19: F821 undefined name '__version__'
./setupegg.py:1:1: F401 'setuptools' imported but unused
```

--- a/.github/workflows/runtests.yml
+++ b/.github/workflows/runtests.yml
@@ -22,13 +22,16 @@ jobs:
         with:
           python-version: '3.10'
       - run: python -m pip install --upgrade pip
-      - run: pip install mpmath flake8 flake8-comprehensions
+      - run: pip install mpmath flake8 flake8-comprehensions ruff
 
       - name: Basic code quality tests
         run: bin/test quality
 
       - name: Run flake8 on the sympy package
         run: flake8 sympy
+
+      - name: Run Ruff on the sympy package
+        run: ruff check .
 
       - name: Detect invalid escapes like '\e'
         run: python -We:invalid -We::SyntaxWarning -m compileall -f -q sympy/

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,0 +1,49 @@
+[tool.ruff]
+# Enable Pyflakes `E` and `F` codes by default.
+select = ["E", "F"]
+
+# Ignore rules that currently fail on the SymPy codebase
+ignore = [
+    "E401",  # Multiple imports on one line
+    "E402",  # Module level import not at top of file
+    "E501",  # Line too long (<LENGTH> > 88 characters)
+    "E701",  # Multiple statements on one line (colon)
+    "E702",  # Multiple statements on one line (semicolon)
+    "E703",  # Statement ends with an unnecessary semicolon
+    "E711",  # Comparison to `None` should be `cond is not None`
+    "E712",  # Comparison to `<BOOL>` should be `cond is <BOOL>`
+    "E713",  # Test for membership should be `not in`
+    "E714",  # Test for object identity should be `is not`
+    "E721",  # Do not compare types, use `isinstance()`
+    "E722",  # Do not use bare `except`
+    "E731",  # Do not assign a `lambda` expression, use a `def`
+    "E741",  # Ambiguous variable name: `<VARIABLE>`
+    "E743",  # Ambiguous function name: `<FUNCTION>`
+    "F401",  # `<TYPE>` imported but unused
+    "F403",  # `from <MODULE> import *` used; unable to detect undefined names
+    "F405",  # `<TYPE>` may be undefined, or defined from star imports: `<MODULE>`
+    "F523",  # `.format` call has unused arguments at position(s): <INDEX>
+    "F601",  # Dictionary key literal `'<KEY>'` repeated
+    "F811",  # Redefinition of unused `<VARIABLE>` from line <LINE>
+    "F821",  # Undefined name `VARIABLE`
+    "F823",  # Local variable `VARIABLE` referenced before assignment
+    "F841",  # Local variable `VARIABLE` is assigned to but never used
+]
+
+# Exclude paths currently excluded in the flake8 configuration
+exclude = [
+    "sympy/parsing/latex/_antlr/*",
+    "sympy/parsing/autolev/_antlr/*",
+    "sympy/parsing/autolev/test-examples/*",
+    "sympy/integrals/rubi/*",
+]
+
+# Black default, although irrelevant with E501 ignored
+line-length = 88
+
+# Allow unused variables when underscore-prefixed.
+dummy-variable-rgx = "^(_+|(_+[a-zA-Z0-9_]*[a-zA-Z0-9]+?))$"
+
+# Per-file ignores currently specified in the flake8 configuration
+[tool.ruff.per-file-ignores]
+"sympy/interactive/session.py" = ["F821"]

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,6 +1,6 @@
 [tool.ruff]
 # Enable Pyflakes `E` and `F` codes by default.
-select = ["E", "F"]
+select = ["C4", "E", "F"]
 
 # Ignore rules that currently fail on the SymPy codebase
 ignore = [
@@ -32,6 +32,8 @@ ignore = [
 
 # Exclude paths currently excluded in the flake8 configuration
 exclude = [
+    "sympy/assumptions/*generated.py",
+    "sympy/core/*_generated.py",
     "sympy/parsing/latex/_antlr/*",
     "sympy/parsing/autolev/_antlr/*",
     "sympy/parsing/autolev/test-examples/*",

